upgrade/upgrade-scripts/upgrade

```diff
@@ -329,16 +331,30 @@ done
 shift $((OPTIND - 1))
 
 case "$1" in
-in-place)
+deferred)
+	UPGRADE_TYPE="${1^^}"
 	shift 1
 	verify_upgrade_is_allowed
-	verify_upgrade_in_place_is_allowed
+    verify_upgrade_in_place_is_allowed
 	upgrade_in_place "$@"
 	;;
-not-in-place)
+full)
+	UPGRADE_TYPE="${1^^}"
 	shift 1
 	verify_upgrade_is_allowed
-	upgrade_not_in_place "$@"
+
+	#
+	# FULL upgrade always perform a reboot but can take on two
+	# flavors -- in-place or not-in-place. The flavor of a FULL upgrade
+	# is determined by the MINIMUM_REBOOT_OPTIONAL_VERSION. If our
+	# upgrade version is not greater than MINIMUM_REBOOT_OPTIONAL_VERSION
+	# than the upgrade will be a not-in-place upgrade.
+	#
+	if is_upgrade_in_place_allowed; then
+		upgrade_in_place "$@"
+	else
+		upgrade_not_in_place "$@"
```

Comment:
Do you think it could be useful to have an option to force `not-in-place` upgrade, at least to make it easier to test that feature?

Reply:
I had discussed this with @prakashsurya yesterday and there is a `-x` option you can use when you `unpack-image` that will force that. So I left a force flag out here since that interface was already being used by QA. I'm happy to add an override.

Reply:
I see. I'm fine with that as long as we have a way. Perhaps add a comment here that mentions that `unpack-image` flag?

Reply:
Regarding the change in the name of the commands, I noticed that upgrade testing succeeded although I haven't seen any related app-gate changes in your ab-pre-push link. Either I'm missing something, or somehow you haven't been testing the right changes?

Reply:
Upgrade testing doesn't use this interface. This will only be used by developers and dcenter upgrade. QA has discussed adding testing for this to ensure that it doesn't break in the future but that doesn't exist right now.

Reply:
I see, thanks for the clarifications.